saffron/src/query.rs

```diff
@@ -0,0 +1,292 @@
+use std::marker::PhantomData;
+
+use crate::utils::decode_from_field_elements;
+use ark_ff::PrimeField;
+use ark_poly::{EvaluationDomain, Evaluations, Radix2EvaluationDomain};
+use rayon::prelude::*;
+use thiserror::Error;
+
+#[derive(Clone, Debug)]
+/// Represents the bytes a user query
+pub struct QueryBytes {
+    pub start: usize,
+    pub len: usize,
+}
+
+#[derive(Copy, Clone, PartialEq, PartialOrd, Eq, Ord, Debug)]
+/// We store the data in a vector of vector of field element
+/// The inner vector represent polynomials
+pub struct FieldElt {
+    /// the index of the polynomial the data point is attached too
+    pub poly_index: usize,
+    /// the index of the root of unity the data point is attached too
+    pub eval_index: usize,
+    domain_size: usize,
+    n_polys: usize,
```

Comment:
I don't understand why we have this field here, if `FieldElt` represents a single field element. Same for domain_size.

Reply:
I used these in order to construct an `Iterator` instance which drastically simplified how we interact with this module. You need to know when to stop a given chunk, or when to stop iterating.